hashring/hashring_test.go

```diff
@@ -302,44 +305,185 @@ func TestLookupN(t *testing.T) {
 	assert.Len(t, unique, 9, "expected to get nine unique servers")
 }
 
-func TestProcessMembershipChanges(t *testing.T) {
-	ring := New(farm.Fingerprint32, 10)
+type ProcessMembershipChangesSuite struct {
+	suite.Suite
+	ring    *HashRing
+	member1 fakeMember
```

Comment:
It smells that we want an array (not a slice) here; `[4]fakeMember` but I am not sure since all tests seem to be a specific order of events on the members. Besides from initialization they don't seem to be operated on in batches.

What were your thought while writing the code and what made you choose for this approach?